engines/router/missionctl/log/resultlog/kafka.go

```diff
@@ -129,11 +129,11 @@ func (l *KafkaLogger) writeToKafka(
 	return nil
 }
 
-func (l *KafkaLogger) write(turLogEntry *TuringResultLogEntry) error {
+func (l *KafkaLogger) write(turLogEntry *turing.TuringResultLogMessage) error {
 	return l.writeToKafka(
```

Comment:
it make sense if its just kafka alone, but for other logger (bq) there isn't a key so changing the `TuringResultLogger` interface to cater to this is a problem.

alternatively the key generation could be moved to this method from `writeToKafka` but I dont have any opinion on this, since it could be also viewed that `writeToKafka` control how the key is generated from Turing across all implementation.

Reply:
I mean to change signature of `writeToKafka` itself since it is only implement kafka, I thought just in case key for `routerLog` and `resultLog` is not the same, you have to change `writeToKafka` in both places

Reply:
hmm its a neat idea but i think that can be done when the key is different, if not it will just be duplicating the same logic in both logger. `writeToKafka` can maintain a consistent way of creating the key till there is a requirement for it to be different